iterative/aws/provider.go

```diff
@@ -188,26 +192,48 @@ func ResourceMachineCreate(ctx context.Context, d *schema.ResourceData, m interf
 	sgID = *sgDesc.SecurityGroups[0].GroupId
 	vpcID = *sgDesc.SecurityGroups[0].VpcId
 
-	subDesc, err := svc.DescribeSubnets(ctx, &ec2.DescribeSubnetsInput{
+	// https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/ec2#Client.DescribeSubnets
+	// default Subnet selection
+	subnetOptions := &ec2.DescribeSubnetsInput{
 		Filters: []types.Filter{
 			{
 				Name:   aws.String("vpc-id"),
 				Values: []string{vpcID},
 			},
 		},
-	})
+	}
+	// use availability zone from user
+	if availabilityZone != "" {
+		subnetOptions.Filters = append(subnetOptions.Filters, types.Filter{
+			Name:   aws.String("availability-zone"),
+			Values: []string{availabilityZone},
+		})
+	}
+	// use exact subnet-id from user
+	if subnetId != "" {
+		subnetOptions.Filters = append(subnetOptions.Filters, types.Filter{
+			Name:   aws.String("subnet-id"),
+			Values: []string{subnetId},
+		})
+	}
+	subDesc, err := svc.DescribeSubnets(ctx, subnetOptions)
 	if err != nil {
 		return decodeAWSError(region, err)
 	}
 	if len(subDesc.Subnets) == 0 {
-		return errors.New("no subnets found")
+		return errors.New("No Subnet found")
```

Comment:
to be consistent with prior 'no ... found' checks

```suggestion
		return errors.New("no Subnet found")
```